modules/aws/sonar-upgrader/python_upgrader/upgrade/main.py

```diff
@@ -269,12 +268,44 @@ def run_test_connection_stage(args, extended_node_dict, upgrade_status_service):
             print(f"### Test connection to all DSF nodes succeeded")
 
 
-def run_collect_python_location_step(args, extended_node_dict, upgrade_status_service):
-    """
-    Collects the Python location in the DSF nodes and fills it in the extended_node_dict
-    """
-    if should_run_python(args):
-        collect_python_locations(extended_node_dict, args.stop_on_failure, upgrade_status_service)
+def run_collect_facts_step(args, extended_node_dict, upgrade_status_service):
+
+    for extended_node in extended_node_dict.values():
+        if upgrade_status_service.should_collect_facts(extended_node.get('dsf_node_id')):
+            upgrade_status_service.update_upgrade_status(extended_node.get('dsf_node_id'),
+                                                         UpgradeStatus.RUNNING_COLLECT_FACTS)
+            try:
+                collect_facts(extended_node)
+                upgrade_status_service.update_upgrade_status(extended_node.get('dsf_node_id'),
+                                                             UpgradeStatus.COLLECT_FACTS_SUCCEEDED)
+            except Exception as e:
+                upgrade_status_service.update_upgrade_status(extended_node.get('dsf_node_id'),
+                                                             UpgradeStatus.COLLECT_FACTS_FAILED)
+                if args.stop_on_failure:
+                    raise UpgradeException(
+                        f"Collecting Python location in {extended_node.get('dsf_node_name')} failed") from e
+                return None
+
+
+def collect_facts(extended_node):
```

Comment:
The name "collect_facts" isn't clear enough, can you change to something like collect_node_info?
Please also change the name of the corresponding UpgradeStatus (in upgrade_status_service.py)

Reply:
@jsonar-cpapke Hi Chris, thank you for changing all the other places to "collect node info", can you please change here too?

Reply:
Should be updated, forgot to push the changes